datafusion/core/tests/physical_optimizer/enforce_sorting.rs

```diff
@@ -1943,6 +1943,30 @@ async fn test_remove_unnecessary_spm1() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+async fn test_remove_unnecessary_spm2() -> Result<()> {
```

Comment:
I think this will be clear if we add multi_partition testing cases? Because SortPreservingMergeExec only need to merge multi partitions?

Reply:
The test says we shouldn't directly remove SPM if it contains `fetch`.  To hit the related code, should be one partition